packages/environment-ember-template-imports/-private/environment/preprocess.ts

```diff
@@ -1,26 +1,86 @@
 import { GlintExtensionPreprocess } from '@glint/core/config-types';
-import { parseTemplates } from 'ember-template-imports';
 import { GLOBAL_TAG, PreprocessData, TemplateLocation } from './common';
-
+/**
+ * We don't need to actually generate the code that would be emitted for
+ * a real app to build, we can use placeholders, so we have less offsets
+ * to worry about (precompile, setComponentTemplate, their imports, etc)
+ */
 const TEMPLATE_START = `[${GLOBAL_TAG}\``;
 const TEMPLATE_END = '`]';
 
+// NOTE: This import is a lie -- We are not real ESM, and this is compiled to CJS.
+//       ESM usage would be different here, and this is the CJS version of content-tag.
+import { Preprocessor } from 'content-tag';
+const p = new Preprocessor();
+
+interface Parsed {
+  type: 'expression' | 'class-member';
+  tagName: 'template';
+  contents: string;
+  range: {
+    start: number;
+    end: number;
+  };
+  contentRange: {
+    start: number;
+    end: number;
+  };
+  startRange: {
+    end: number;
+    start: number;
+  };
+  endRange: {
+    start: number;
+    end: number;
+  };
+}
+
 export const preprocess: GlintExtensionPreprocess<PreprocessData> = (source, path) => {
-  let templates = parseTemplates(source, path, { templateTag: 'template' }).filter(
-    (match) => match.type === 'template-tag'
-  );
+  // NOTE: the preprocessor requires fully syntactically correct code
+  //       which isn't appropriate for an LSP, so
+  //       1. we need to catch parse errors
+  //       2. we need to move to something else, so we can have
+  //          intellisense on in-progress-typed code (such as `@`)
```

Comment:
It sounds like this comment isn't entirely accurate (though catching the error this way seems like the right thing to do regardless)

Reply:
removed :upside_down_face: